dagrunner/utils/networkx.py

```diff
@@ -144,18 +192,31 @@ def _collapse_graph(graph: nx.DiGraph, collapse_properties: Union[str, Iterable[
             except TypeError:
                 continue
 
-        node_data_lookup[node] = set(
-            [
-                str(
-                    {
-                        key: val
-                        for key, val in graph.nodes(data=True)[gnode].items()
-                        if key not in collapse_properties
-                    }
-                )
-                for gnode in filtered_nodes
-            ]
-        )
+        if collapsed_data_summary:
```

Comment:
For visualising huge graphs, the visualisation was fine but the html table itself contained huge amounts.  In this case, a more summarised representation of collapsed data is preferred (perhaps we make default/the only option at some point).